### DISCUSSION TITLE
replacement for wpf trigger (applied to ListBoxItem.IsSelected property)

### DISCUSSION THREAD

**Discussion by jmkinzer** [AvaloniaUI/Avalonia#14086] [General]
Hi, in WPF I have something like the following:
```
<ListBox>
    <ListBox.ItemContainerStyle>
        <Style TargetType="ListBoxItem">
            <Style.Triggers>
                <Trigger Property="IsMouseOver" Value="True">
                    <Setter Property="IsSelected" Value="True"/>
                </Trigger>
            </Style.Triggers>
        </Style>
    </ListBox.ItemContainerStyle>
</ListBox>
```
In Avalonia I'm able to use a selector as follows but then the issue is that IsSelected refers to the ContentPresenter, not the ListBoxItem. So either I need a way to access the parent here or a different approach?
```
        <ControlTheme TargetType="ListBoxItem" BasedOn="{StaticResource {x:Type ListBoxItem}}">
          <!--  PointerOver State  -->
          <Style Selector="^:pointerover /template/ ContentPresenter#PART_ContentPresenter">
            <!-- problem: this refers to the ContentPresent not the ListBoxItem -->
            <Setter Property="IsSelected" Value="True"/>
          </Style>
        </ControlTheme>
```
I'd really appreciate any hints here. Thanks!

**Comment by maxkatz6** [comment]
> In Avalonia I'm able to use a selector as follows but then the issue is that IsSelected refers to the ContentPresenter

You don't need selector to target ContentPresenter. Simpler "^:pointerover" would take ListBoxItem and apply setter on it.

Or, even simpler would be this way, without redefining Control Theme:
```xaml
<ListBox>
    <ListBox.Styles>
        <Style Selector="ListBoxItem:selected">
            <Setter Property="IsSelected" Value="True"/>
        </Style>
    </ListBox.Styles>
</ListBox>
```

**Reply by maxkatz6** [reply]
Though, there is another problem, as far as I remember, IsSelected property is used by the virtualization algorithm, and it overrides these properties with a local value, making it unusable from styles. Can't find the issue.

**Reply by rabbitism** [reply]
https://github.com/AvaloniaUI/Avalonia/issues/10249

**Comment by jmkinzer** [comment]
Thanks - Yes, I had tried that but while it does indeed select the very first time one hovers over an item the selection will not subsequently change. From the bug report it sounds like this really isn't the way to approach this...(?) but I do require the controltheme and I wasn't sure how else to go about it...
